File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.java
class Solution {
    public int findMin(int[] nums) {
        int lefti = 0;
        int righti = nums.length - 1;
        int midi = nums.length / 2;

        if (nums.length == 1){
            return nums[0];
        }
        
        while(true){
            // if the number right of mid is smaller, right is smallest
            if(nums[midi] > nums[(midi+1)%nums.length]){
                return nums[(midi+1)%nums.length];
            }
            // if the number left of mid is bigger, mid is smallest
            else if(nums[midi-1 < 0 ? nums.length - 1 : midi-1] > nums[midi]){
                return nums[midi];
            }
            // this is a sorted segment on the right
            // so the split must be left
            if (nums[midi] < nums[righti]){
                righti = midi;
                midi = (lefti + righti) / 2;
            }
            // sorted on the left
            else if (nums[midi] > nums[lefti]){
                lefti = midi;
                midi = (lefti + righti) / 2;
            }
        }
        
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        pos = {}
        maxx = 0
        for i, n in enumerate(nums):
            if n in pos:
                continue
            elif n+1 in pos and n-1 in pos:
                left = pos[n-1][0]
                right = pos[n+1][1]
            elif n-1 in pos:
                left = pos[n-1][0]
                right = n
            elif n+1 in pos:
                right = pos[n+1][1]
                left = n
            else:
                right = n
                left = n
            
            maxx = max(maxx, right-left+1)
            pos[left] = (left, right)
            pos[right] = (left, right)
        return maxx





        